Surface config extensions package in Options topic

The following sample code does not compile on `2.1.300`:

```csharp
IServiceCollection services = new ServiceCollection();
IConfiguration config = new ConfigurationBuilder().Build();
services.Configure<MyOptions>("named_options_1", config);
```

The following error is given:

```
error CS1503: Argument 3: cannot convert from 'Microsoft.Extensions.Configuration.IConfiguration' to 'System.Action<MyOptions>'
```

The only available `Configure` extension methods have the following signatures:

```csharp
public static IServiceCollection Configure<TOptions>(this IServiceCollection services, string name, Action<TOptions> configureOptions) where TOptions : class;
public static IServiceCollection Configure<TOptions>(this IServiceCollection services, Action<TOptions> configureOptions) where TOptions : class;
```

Neither of which, as you can see, take an `IConfiguration` as a parameter which is what the [documentation](https://github.com/aspnet/Docs/blob/a9bbe631355235df6c773d5b1ad08dc05f8ed62b/aspnetcore/fundamentals/configuration/options/sample/Startup.cs#L64) tells readers to provide.

---

Note that samples 1, 3, and 6 do not use the signatures reproduced above so I assume they are all invalid.

---

This is quite frustrating as a user who is now spending time on wiring up configuration instead of writing and validating business logic.

---

Is there a specific NuGet package that will provide the exact signatures used in the samples that is not one of the following (locally all of these are version `2.1.1`)?

```
Microsoft.Extensions.Configuration
Microsoft.Extensions.Configuration.EnvironmentVariables
Microsoft.Extensions.Configuration.Json
Microsoft.Extensions.DependencyInjection
Microsoft.Extensions.Options
```

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: ed39cb0e-aa85-8bba-ffb4-859b56a60723
* Version Independent ID: dc7e3449-650b-d117-5eac-69e5ccb250cb
* Content: [Options pattern in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-2.1)
* Content Source: [aspnetcore/fundamentals/configuration/options.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/fundamentals/configuration/options.md)
* Product: **aspnet-core**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

--- a/aspnetcore/fundamentals/configuration/options.md
+++ b/aspnetcore/fundamentals/configuration/options.md
@@ -18,6 +18,26 @@ The options pattern uses classes to represent groups of related settings. When c
 
 [View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/fundamentals/configuration/options/sample) ([how to download](xref:tutorials/index#how-to-download-a-sample)) This article is easier to follow with the sample app.
 
+## Prerequisites
+
+::: moniker range=">= aspnetcore-2.1"
+
+Reference the [Microsoft.AspNetCore.App metapackage](xref:fundamentals/metapackage-app) or add a package reference to the [Microsoft.Extensions.Options.ConfigurationExtensions](https://www.nuget.org/packages/Microsoft.Extensions.Options.ConfigurationExtensions/) package.
+
+::: moniker-end
+
+::: moniker range="= aspnetcore-2.0"
+
+Reference the [Microsoft.AspNetCore.All metapackage](xref:fundamentals/metapackage) or add a package reference to the [Microsoft.Extensions.Options.ConfigurationExtensions](https://www.nuget.org/packages/Microsoft.Extensions.Options.ConfigurationExtensions/) package.
+
+::: moniker-end
+
+::: moniker range="< aspnetcore-2.0"
+
+Add a package reference to the [Microsoft.Extensions.Options.ConfigurationExtensions](https://www.nuget.org/packages/Microsoft.Extensions.Options.ConfigurationExtensions/) package.
+
+::: moniker-end
+
 ## Basic options configuration
 
 Basic options configuration is demonstrated as Example &num;1 in the [sample app](https://github.com/aspnet/Docs/tree/master/aspnetcore/fundamentals/configuration/options/sample).

